agenta-backend/agenta_backend/routers/evaluation_router.py

```diff
@@ -12,18 +12,19 @@
     NewEvaluation,
     DeleteEvaluation,
     EvaluationType,
-    EvaluationStatus,
+    EvaluationUpdate,
 )
 from agenta_backend.services.results_service import (
     fetch_results_for_human_a_b_testing_evaluation,
     fetch_results_for_auto_exact_match_evaluation,
     fetch_results_for_auto_similarity_match_evaluation,
+    fetch_results_for_auto_regex_test,
     fetch_results_for_auto_ai_critique,
 )
 from agenta_backend.services.evaluation_service import (
     UpdateEvaluationScenarioError,
     update_evaluation_scenario,
-    update_evaluation_status,
+    update_evaluation,
```

Comment:
I wouldn't use the same method for all the updates as it will end up with a very loong method.
I would use a separate method to update the evaluation_type_settings

Reply:
The issue is if I use a separate service for update settings, then firstly, there will be 2 db calls made to the same document. Secondly, the API should return a consolidated result of the evaluation after the update. But, these 2 separate services will return different evaluation objects. This could be solved by using a different approach though, but will require some amount of rewriting of the router and services. Let me know your thoughts.

Reply:
I disagree @aakrem I think we should use one method to update evaluation status/*. There is no reason that this method should grow big. At the end of the day, the method should only do CRUD operations. 
The current issue is that update_evaluation_scenario for instance does something other than the update (it computes evaluate for AI critic) which makes it long.